server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -715,47 +722,54 @@ Batch nextBatch(SortedKeyValueIterator<Key,Value> iter, Range range, int num, Se
 
     long maxResultsSize = tableConfiguration.getAsBytes(Property.TABLE_SCAN_MAXMEM);
 
-    if (columns.size() == 0) {
-      iter.seek(range, LocalityGroupUtil.EMPTY_CF_SET, false);
-    } else {
-      iter.seek(range, LocalityGroupUtil.families(columns), true);
-    }
-
     Key continueKey = null;
     boolean skipContinueKey = false;
 
     boolean endOfTabletReached = false;
-    while (iter.hasTop()) {
 
-      value = iter.getTopValue();
-      key = iter.getTopKey();
+    try {
+      if (columns.size() == 0) {
+        iter.seek(range, LocalityGroupUtil.EMPTY_CF_SET, false);
```

Comment:
If `seek` throws the `ScanYieldException` and there exists a Key in the data which is the same as the startKey of the Range, I believe we would miss data.

`skipContinueKey` is set to true (in the catch below), but when we're `seek()`'ing here, we haven't yet consumed that top key. Thus, on a re-seek, we would skip past it.

My hunch is that you need a two catch statements: one here that does not skip the continueKey and the one below around the `while(hasTop)` (as-is).